### PR TITLE
Fix tests to mostly run in new luvi

### PR DESCRIPTION
**Pull request by creationix**
This branch is for testing luvit with the new in-progress luvi release.

The updates to lua-openssl seems to have broken some stuff.

To test this, use latest luvi from master instead of stable luvi.

### CODE REVIEW

File: deps/tls/common.lua [hunk 1]
```diff
@@ -393,9 +393,9 @@ end
 exports.TLSSocket = TLSSocket
 
 -------------------------------------------------------------------------------
-local VERIFY_PEER = { "peer" }
-local VERIFY_PEER_FAIL = { "peer", "fail_if_no_peer_cert" }
-local VERIFY_NONE = { "none" }
+local VERIFY_PEER = openssl.ssl.peer
+local VERIFY_PEER_FAIL = openssl.ssl.fail
```
Comment (zhaozg) [inline deps/tls/common.lua]: hi, ssl.peer can be bit combine with ssl.fail or ssl.once, VERIFY_PEER_FAIL should be ssl.fail+ssl.peer(bit.bor(ssl.fail,ssl.peer)

**Comment by zhaozg** [comment]
please see this. https://github.com/luvit/luvit/pull/819

**Comment by creationix** [comment]
Thanks, I postponed the lua-openssl bump in luvi for now, but this will be helpful for the next luvi release.

**Comment by zhaozg** [comment]
#819 with something wrong, so I make a new PR https://github.com/luvit/luvit/pull/848,

change note:

   lua-openssl object use obj:tostring() replace tostring(obj)
   sslctx verify mode change from table to number
   default will not set global openssl variable

**Pull request closed without merging**